modules/nw-http-header-configuration.adoc

```diff
@@ -0,0 +1,135 @@
+// Module included in the following assemblies:
+//
+// * networking/ingress-operator.adoc
+// * networking/route-configuration.adoc
+
+:_content-type: CONCEPT
+[id="nw-http-header-configuration_{context}"]
+= HTTP header configuration
+
+{product-title} provides a different methods for working with HTTP headers. When setting or deleting headers, you can use specific fields in the Ingress Controller or an individual route to modify request and response headers. Certain headers can also be set using route annotations. The various ways of configuring headers can present challenges when working together.
+
+[NOTE]
+====
+Headers can only be set or deleted within an `IngressController` or `Route` CR, they cannot be appended. If an HTTP header is set with a value, that value is expected to be complete and not require appending in the future. In situations where it makes sense to append a header, such as the X-Forwarded-For header, use the `spec.httpHeaders.forwardedHeaderPolicy` field, instead of `spec.httpHeaders.actions`.
+====
+
+== Order of precedence
+
+When the same HTTP header is modified both in the Ingress Controller and route object, HAProxy prioritizes the actions in certains ways depending on if it is a request or response header.
+
+* In the case of HTTP response headers, actions specified in the Ingress Controller are executed after the actions specified in a route. This means that the actions specified in the Ingress Controller take precedence.
+
+* In the case of HTTP request headers, actions specified in a route object are executed after the actions specified in the Ingress Controller. This means that the actions specified in the route object take precedence.
+
+For example, if a cluster admin sets the X-Frame-Options response header with the value `DENY` in the Ingress Controller:
+
+.Example Ingress Controller spec
+[source,yaml]
+----
+apiVersion: operator.openshift.io/v1
+kind: IngressController
+...
+spec:
+  httpHeaders:
+    actions:
+      response:
+      - name: X-Frame-Options
+        action:
+          type: Set
+          set:
+            value: DENY
+----
+
+And if a route owner sets the same X-Frame-Options response header with the value `SAMEORIGIN`:
+
+.Example route spec
+[source,yaml]
+----
+apiVersion: route.openshift.io/v1
+kind: Route
+...
+spec:
+  httpHeaders:
+    actions:
+      response:
+      - name: X-Frame-Options
+        action:
+          type: Set
+          set:
+            value: SAMEORIGIN
+----
+
+Then the value set for the X-Frame-Options header at the global level in the Ingress Controller will take precedence, even if a specific route allows frames.
```

Comment:
Same about not trying to continue this sentence from the above ones. Should be complete on its own